libcaf_core/caf/flow/step/take_last.hpp

```diff
@@ -0,0 +1,58 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#pragma once
+
+#include "caf/detail/ring_buffer.hpp"
+#include "caf/fwd.hpp"
+
+#include <cstddef>
+
+namespace caf::flow::step {
+
+template <class T>
+class take_last {
+public:
+  using input_type = T;
+
+  using output_type = T;
+
+  explicit take_last(size_t num)
+    : last_elements_count_(num),
+      elements_(caf::detail::ring_buffer<output_type>(num)) {
```

Comment:
Why are you creating a temporary to initialize this member?

```suggestion
      elements_(num) {
```